spec/lita/handlers/github_prs/git_diff_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+RSpec.describe Lita::Handlers::GithubPrs::GitDiff do
+  describe '#modified_files' do
+    it 'returns a list of modified files' do
+      files = [
+        build_file('modified_file.rb', status: 'modified'),
+        build_file('added_file.rb', status: 'added'),
+        build_file('deleted_file.rb', status: 'deleted')
+      ]
+      diff = build_diff(files: files)
+      git_diff = Lita::Handlers::GithubPrs::GitDiff.new(diff)
+
+      expect(git_diff.modified_files).to eq(['modified_file.rb'])
+    end
+  end
+
+  describe '#added_files' do
+    it 'returns a list of added files' do
+      files = [
+        build_file('modified_file.rb', status: 'modified'),
+        build_file('added_file.rb', status: 'added'),
+        build_file('deleted_file.rb', status: 'deleted')
+      ]
+      diff = build_diff(files: files)
+      git_diff = Lita::Handlers::GithubPrs::GitDiff.new(diff)
+
+      expect(git_diff.added_files).to eq(['added_file.rb'])
+    end
+  end
+
+  describe '#pull_requests' do
+    it 'returns all merge commits from pull requests' do
+      merge_commit = build_commit('Merge pull request: Fix world hunger')
+      commits = [
+        build_commit('Fiddle with fiddlesticks'),
+        merge_commit,
+        build_commit('Fix world hunger')
+      ]
+      diff = build_diff(commits: commits)
+      git_diff = Lita::Handlers::GithubPrs::GitDiff.new(diff)
+
+      expect(git_diff.pull_requests).to eq([merge_commit.commit])
+    end
+  end
+
+  def build_file(name, status:)
+    double(
+      filename: name,
+      status: status
+    )
+  end
+
+  def build_commit(message)
+    double(
+      commit: double(
+        message: message
+      )
+    )
+  end
+
+  def build_diff(files: [], commits: [])
+    double(
+      files: files,
+      commits: commits
+    )
+  end
```

Comment:
I think that here I *might* argue against abstracting away this into methods.  Particularly with `build_file` there isn't much difference between `build_file('foo.rb', status: 'modified`)` and `double(filename: 'foo.rb', status: 'modified')`, but the latter makes it clear in the test that you're dealing with a `double`.  I could see how it would be a bit more painful with `build_commit`, though.

Though each definition is different, so it maybe doesn't hurt to abstract it away into a somewhat consistent set of builder functions.  Maybe it depends on how much we expect this to grow.  Since it's a pretty simple class I suppose it's not likely to grow much.

One thing that I thought would be good to test is when you have an empty array of files, but it occurs to me that if you're comparing diffs that you should always have at least one file, so that's probably not needed...

Reply:
I agree that it isn't that great to hide the fact that doubles are used. They are there simply because the actual returned value is a `Sawyer` object and I didn't want to use that since it's an implementation detail of a dependency :/. I considered using `OpenStruct` but felt like doubles would be good enough as the only requirement is that the values should be accessible with dot notations. Do you know of a better option for that?

For now I don't expect this to grow much but in an optimal world it would be good to have a factory with the full document to make sure we don't hide anything from anyone wanting to implement features based on other available values.

Seems like that test couldn't hurt even though it "shouldn't be needed". :)